x86/SearchMacros.asm

```diff
@@ -354,54 +354,62 @@ Display	2, "Search(alpha=%i1, beta=%i2,	depth=%i8, cutNode=%i9)	called%n"
 		cmp   esi, dword[.evalu]
 		 jg   .8skip
 		add   esi, 225
-    if USE_MATEFINDER =	0
-	      movzx   ecx, word[rbx+State.npMaterial+2*rcx]
-	       test   ecx, ecx
-		 jz   .8skip
-    else
-		mov   r8d, dword[.evalu]
-		mov   ecx, dword[rbx+State.npMaterial]
-	       test   ecx, 0x0FFFF
-		 jz   .8skip
-		shr   ecx, 16
-		 jz   .8skip
-		add   r8d, 2*VALUE_KNOWN_WIN-1
-		cmp   r8d, 4*VALUE_KNOWN_WIN-1
-		jae   .8skip
-    end	if
 		cmp   eax, esi
 		 jl   .8skip
-
-    if USE_MATEFINDER =	1
-		mov   edx, dword[.depth]
-		cmp   edx, 4
-		jbe   .8do
-		sub   rsp, MAX_MOVES*sizeof.ExtMove
-		mov   rdi, rsp
-	       call   Gen_Legal
-		xor   ecx, ecx
-		xor   eax, eax
-		mov   rdx, rsp
-		cmp   rdx, rdi
-		jae   .8loopdone
-    .8loop:
-		mov   r8d, [rdx+ExtMove.move]
-		shr   r8d, 6
-		and   r8d, 63
-		cmp   byte[rbp+Pos.board+r8], King
-	       sete   r8l
-		add   ecx, r8d
-		add   rdx, sizeof.ExtMove
-		add   eax, 1
-		cmp   rdx, rdi
-		 jb   .8loop
-    .8loopdone:
-		add   rsp, MAX_MOVES*sizeof.ExtMove
-	       test   ecx, ecx
-		 jz   .8skip
-		cmp   eax, 6
-		 jb   .8skip
-    end	if
+		; && (ss->ply >= nmp_ply || ss->ply % 2 == pair)) 
+		
+		mov   r12b, byte[rbx-1*sizeof.State+State.ply]
+		mov   dl, byte[rbp-Thread.rootPos+Thread.nmp_ply]
+		cmp   r12b, dl
+		jge   .8do
+		
+		mov   r12b, byte[rbx-1*sizeof.State+State.ply]
+		mov   dl, byte[rbp-Thread.rootPos+Thread.pair]
+		and   r12b, 1 ; ss->ply % 2
+		cmp   r12b, dl
+		jne   .8skip
+		
+    ; if USE_MATEFINDER =	1
```

Comment:
Commented out MateFinder sections to avoid confusing myself. I will restore them in the final patch.